### DISCUSSION TITLE
Should TypeScript be a devDependency?  (It now shows up under dependencies...)

### DISCUSSION THREAD

**Discussion by jasonkylefrank** [vercel/next.js#44283] [Help]
After running `npx create-next-app@latest` to generate a project, I now see that TypeScript (and some related things like `@types/node`, `@types/react`, `@types/react-dom`) are listed under `dependencies` in the `package.json`.  

To be clear, the TypeScript dependencies are *not* listed under `devDependencies`.

I'm still somewhat new to TypeScript, but I would have thought that TypeScript would be a `devDependency`?

Furthermore, if TypeScript is listed as a `dependency`, instead of a `devDependency` does that mean that the JavaScript bundle that gets sent to the client is larger than it should be?

Thanks for the help.

**Comment by boar-is** [comment]
[Related](https://github.com/vercel/next.js/discussions/43277)

**Reply by boar-is** [reply]
> [Related](https://github.com/vercel/next.js/discussions/43277)

Anyway, I also want to decide (once and for all), can I just put everything into `dependencies` and forget about `devDependencies`?

**Reply by ijjk** [reply]
Yeah as the linked thread mentions the separation of `dependencies` and `devDependencies` is no longer required with `output: standalone` support which ensures actual separation between necessary files and un-necessary which `devDependencies` did not achieve.

**Reply by boar-is** [reply]
As I understand from [here](https://nextjs.org/docs/advanced-features/output-file-tracing#automatically-copying-traced-files), `output: standalone` is an opt-in option (not default). But how about fresh `npx create-next-app`? Or is it applies to `output: standalone` only?

**Reply by jasonkylefrank** [reply]
Hi all.  Can you clarify/summarize this?  

I don't understand the `output: standalone` issue and my `next.config.mjs` file does not have that setting in it currently.  Is it supposed to?   Was I supposed to know that I should have added it?   

And I'm still not clear on whether or not my JavaScript bundle which gets sent to the client will be larger than it should be since TypeScript is currently listed as a `dependency` (instead of a `devDependency`)?

Thanks in advance for clarifying/answering!  🙌
